cmd/frontend/internal/cli/config.go

```diff
@@ -445,17 +445,20 @@ func (c *configurationSource) Read(ctx context.Context) (conftypes.RawUnified, e
 	}
 
 	return conftypes.RawUnified{
+		ID:                 site.ID,
 		Site:               site.Contents,
 		ServiceConnections: serviceConnections(c.logger),
 	}, nil
 }
 
-func (c *configurationSource) Write(ctx context.Context, input conftypes.RawUnified) error {
-	// TODO(slimsag): future: pass lastID through for race prevention
+func (c *configurationSource) Write(ctx context.Context, input conftypes.RawUnified, lastID int32) error {
 	site, err := c.db.Conf().SiteGetLatest(ctx)
 	if err != nil {
 		return errors.Wrap(err, "ConfStore.SiteGetLatest")
 	}
+	if site.ID != lastID {
+		return errors.New("last site ID does not match latest site ID")
```

Comment:
I think this error is not very useful to the consumer, especially from what I've seen on the UI.
I'd suggest the following:
```suggestion
		return errors.New("site config has been modified by another request, write not allowed")
```

Reply:
Thanks, I agree. I was trying to think of a way to say what you suggested but my brain did not want to language